roboteam_ai/src/skills/ReflectKick.cpp

```diff
@@ -0,0 +1,115 @@
+//
+// Created by robzelluf on 4/9/19.
+//
+
+#include "ReflectKick.h"
+
+namespace rtt {
+namespace ai {
+
+ReflectKick::ReflectKick(string name, bt::Blackboard::Ptr blackboard)
+        :Skill(std::move(name), std::move(blackboard)) {
+}
+
+void ReflectKick::onInitialize() {
+    auto field = world::field->get_field();
+    goalTarget = getFarSideOfGoal();
+    reflectionPos = robot->pos;
+    numTreeGtp.setAvoidBall(false);
+}
+
+ReflectKick::Status ReflectKick::onUpdate() {
+    angleToGoalTarget = (goalTarget - getKicker()).toAngle();
```

Comment:
This can be split in more functions and be easier to read. It's hard to follow what you're doing.

Reply:
It's hard to follow where you are doing the mathematics that is unique (e.g. the angles that are different from normal intercept), maybe make a small comment or put them in one function